src/common/src/catalog/column.rs

```diff
@@ -243,66 +205,46 @@ impl ColumnDesc {
     /// If the type is atomic, it returns simply itself.
     /// If the type has multiple nesting levels, it traverses for the tree-like schema,
     /// and returns every tree node.
+    ///
+    /// The name will reflect the full path of the column, and the id will be the placeholder.
+    /// Should only be used for observability/debugging purposes.
     pub fn flatten(&self) -> Vec<ColumnDesc> {
```

Comment:
> Should only be used for observability/debugging purposes.

I think for these we should always name it like `xxx_for_debug` to avoid misuse. Because developers don't always read docs. 🤣

Reply:
Currently, there is only one caller in the handler for `DESCRIBE table`. I might update it in the future to stop returning `ColumnDesc`, which would prevent potential misuse.

Reply:
Makes sense. Done in [`9f9565d` (#20497)](https://github.com/risingwavelabs/risingwave/pull/20497/commits/9f9565dfdfdf7f5d4d2620e1cd98175e747299ca).